gensim/sklearn_api/d2vmodel.py

```diff
@@ -159,7 +164,8 @@ def fit(self, X, y=None):
             The trained model.
 
         """
-        if isinstance([i for i in X[:1]][0], doc2vec.TaggedDocument):
+        
+        if isinstance(self._get_first(X), doc2vec.TaggedDocument):
```

Comment:
What happened to the `next(iter(x))` pattern? Is it not suitable here?

Reply:
It doesn't work on Pandas dataframe:


```python
>>> a = pd.DataFrame([[1,2], [2,4]], columns=['1', '2'])
>>> a.index += 1
>>> _get_first = lambda X: next(iter(X))
>>> _get_first(a)
'1'```

Reply:
That's weird. So `next(iter(x))` returns something else than `list(X)[0]`, on non-empty inputs?

If so, I bet other things will be broken too.

Reply:
Pandas has special .iterrows() method for that, I guess it is there for reasons)